packages/desktop/views/onboarding/views/app-setup/views/WelcomeView.svelte

```diff
@@ -1,38 +1,61 @@
     import { OnboardingLayout } from '@components'
-    import features from '@features/features'
-    import { mobile } from '@core/app'
+    import {
+        PRIVACY_POLICY_URL,
+        PRIVACY_POLICY_VERSION,
+        TERMS_OF_SERVICE_URL,
+        TERMS_OF_SERVICE_VERSION,
+    } from '@core/app/constants'
+    import { lastAcceptedPrivacyPolicy, lastAcceptedTermsOfService } from '@core/app/stores'
+    import { openUrlInBrowser } from '@core/app/utils'
     import { localize } from '@core/i18n'
-    import { formatProtocolName, NetworkProtocol } from '@core/network'
+    import { NetworkProtocol, formatProtocolName } from '@core/network'
     import { appSetupRouter } from '@core/router'
+    import features from '@features/features'
+    import { Animation, Button, Checkbox, Text, TextType, Link } from '@ui'
+
+    let termsAccepted: boolean = false
+
+    function onTermsOfServiceClick(): void {
+        openUrlInBrowser(TERMS_OF_SERVICE_URL)
+    }
+
+    function onPrivacyPolicyClick(): void {
+        openUrlInBrowser(PRIVACY_POLICY_URL)
+    }
 
     function onContinueClick(): void {
+        lastAcceptedTermsOfService.set(TERMS_OF_SERVICE_VERSION)
+        lastAcceptedPrivacyPolicy.set(PRIVACY_POLICY_VERSION)
         $appSetupRouter.next()
     }
 </script>
 
 <OnboardingLayout allowBack={false}>
     <div slot="leftpane__content">
-        <div class="flex flex-col {$mobile && 'items-center text-center px-10'} space-y-4 mb-8">
-            {#if !$mobile}
-                <Logo width="64px" logo="logo-firefly" classes="mb-6" />
-            {/if}
-            <Text type={$mobile ? 'h3' : 'h1'}
-                >{localize('views.onboarding.appSetup.welcome.title', {
-                    values: {
-                        protocol: features?.onboarding?.iota?.enabled
-                            ? formatProtocolName(NetworkProtocol.IOTA)
-                            : formatProtocolName(NetworkProtocol.Shimmer),
-                    },
-                })}
-            </Text>
-        </div>
+        <Text type={TextType.h1}
+            >{localize('views.onboarding.appSetup.welcome.title', {
+                values: {
+                    protocol: features?.onboarding?.iota?.enabled
+                        ? formatProtocolName(NetworkProtocol.IOTA)
+                        : formatProtocolName(NetworkProtocol.Shimmer),
+                },
+            })}
+        </Text>
     </div>
-    <div slot="leftpane__action">
-        <Button onClick={onContinueClick} classes="w-full">{localize('actions.continue')}</Button>
+    <div slot="leftpane__action" class="flex flex-col space-y-8">
+        <Checkbox bind:checked={termsAccepted}>
+            <Text slot="label" type={TextType.p} secondary>
+                I've read and I accept the <Link onClick={onTermsOfServiceClick}>Terms of Service</Link> and <Link
+                    onClick={onPrivacyPolicyClick}>Privacy Policy</Link
+                >
```

Comment:
Shouldn't this be localized?

Reply:
As discussed on Discord, we don't know the locale at the start-up, so we have to rethink the localization of the welcome page if we want to support more languages.